pkg/file/lazy_read_closer.go

```diff
@@ -7,6 +7,16 @@ import (
 )
 
 var _ io.ReadCloser = (*LazyReadCloser)(nil)
+var _ LazyReader = (*LazyReadCloser)(nil)
+
+// LazyReader is a single interface with all reading functions we want to use
+// across clients. The transition from io.ReadCloser should happen gradually.
+type LazyReader interface {
```

Comment:
why is there a new exported interface needed?

Reply:
I tried it without a new interface, but then the golang cataloger had to cast the reader for each different function call. 
With an exported interface it casts only once -- this interface covers all internal needs.